OpenSim/Common/Exception.h

```diff
@@ -47,7 +47,31 @@
 //=============================================================================
 //=============================================================================
 
-namespace OpenSim { 
+namespace OpenSim {
+
+
+/** Macros to throw OpenSim exceptions. The purpose of these macros is to avoid
+having to provide the first few arguments (which are common to all OpenSim
+exceptions) to the exception constructor in the throw statements. This also
+allows us to add more details (eg class name) later easily.                   
+
+Note -- Extra braces enclosing "if" are to avoid problems when these macros are
+called within if-else statements like:
+\code{.cpp}
+          if(<some condition>)
+              OPENSIM_THROW_IF(<arguments>)
+          else
+              <some statements>                                               
+\endcode                                                                      */
+#define OPENSIM_THROW_IF(CONDITION, EXCEPTION, ...)                  \
+    {                                                                \
+    if(CONDITION)                                                    \
+        throw EXCEPTION{__FILE__, __LINE__, __func__, __VA_ARGS__};  \
+    }
+
+#define OPENSIM_THROW(EXCEPTION, ...)                                \
+    OPENSIM_THROW_IF(true, EXCEPTION, __VA_ARGS__)
+
 
```

Comment:
Another reason to do this as Ajay suggests is that it is very common when formatting a message generated by a conditional exception to include a "stringized" version of the condition that failed in the message. We wouldn't want to include "true" in the unconditional message. Simbody's messages work that way -- typically they spit out the message and follow it with text like `(condition "i < maxSize" failed)`; not sure if that would be appropriate for OpenSim messages but it would be good to have the option.